src/coreComponents/rajaInterface/GEOS_RAJA_Interface.hpp

```diff
@@ -98,6 +150,26 @@ template< typename POLICY >
 using AtomicPolicy = typename internalRajaInterface::PolicyMap< POLICY >::atomic;
 
 
+RAJA_INLINE bool testAllDeviceEvents( parallelDeviceEvents & events )
+{
+  bool allDone = true;
+  for( auto & event : events )
+  {
+    if( !event.check() )
+    {
+      allDone = false;
+      break;
+    }
+  }
+  return allDone;
+}
+
+RAJA_INLINE void waitAllDeviceEvents( parallelDeviceEvents & events )
+{
+  // poll device events for completion then wait 10 nanoseconds 6,000,000,000 times (60 sec timeout)
+  // 10 nsecs ~= 30 clock cycles @ 3Ghz
+  GEOSX_ASYNC_WAIT( 6000000000, 10, testAllDeviceEvents( events ) );
```

Comment:
Why not just wait on each event in a loop? With CUDA events this will call `cudaEventSynchronize(m_event)`, which should be much more efficient (possibly even waiting for an OS signal). The only downside is that we don't have a timeout period, but I think that's more than worth the tradeoff.

Reply:
Ah yeah using builtin blocking waits in the waitall would probably be better. I probably just defaulted to building the blocking on top of the nonblocking implementation.

I'll probably put that in the follow-up which has a couple other items for cleanup as well.